theseus/geometry/functional/so3.py

```diff
@@ -70,6 +70,11 @@ def check_left_act_matrix(matrix: torch.Tensor):
         raise ValueError("Inconsistent shape for the matrix.")
 
 
+def check_left_project_matrix(matrix: torch.Tensor):
+    if matrix.shape[-2:] != (3, 3):
+        raise ValueError("Inconsistent shape for the matrix.")
```

Comment:
Maybe less ambiguity in the error message about what is 'Inconsistent' would be helpful.

Reply:
Agreed. You can mention what the expected shape is.